### ISSUE TITLE
SEC004: Detect DROP TABLE/DATABASE statements

### ISSUE THREAD

**Issue by RAprogramm** [RAprogramm/sql-query-analyzer#5]
## Summary

Add a rule to detect `DROP TABLE` and `DROP DATABASE` statements — the most destructive DDL operations in SQL.

## Why This Matters

DROP operations are irreversible and catastrophic:

- **Permanent data loss** — no undo, no rollback after commit
- **Schema destruction** — entire table structure is removed
- **Cascading effects** — foreign key relationships, views, and stored procedures may break
- **No confirmation** — executes immediately without warning

### Real-World Incidents

- [Junior dev mass-deletes production database](https://www.reddit.com/r/cscareerquestions/comments/6ez8ag/accidentally_destroyed_production_database_on/) — famous Reddit story
- [Pixar's Toy Story 2 deletion](https://thenextweb.com/news/how-pixars-toy-story-2-was-deleted-twice-once-by-technology-and-once-by-humans) — `rm -rf` equivalent incident
- Common in CI/CD pipelines when environment variables point to wrong database

## Rule Specification

| Property | Value |
|----------|-------|
| Rule ID | `SEC004` |
| Name | DROP statement detected |
| Category | Security |
| Severity | Error |
| Default | Enabled |

### Detection Patterns

```sql
-- Should trigger (Error severity)
DROP TABLE users;
DROP TABLE IF EXISTS orders;
DROP DATABASE production;
DROP SCHEMA public CASCADE;

-- Lower severity (Warning) - might be intentional in migrations
DROP INDEX idx_users_email;
DROP VIEW user_summary;
```

### Suggested Messages

- DROP TABLE/DATABASE: "DROP TABLE/DATABASE permanently destroys data and schema. Ensure this is intentional and backups exist."
- DROP INDEX/VIEW: "DROP INDEX/VIEW detected. Verify this won't impact query performance."

## References

- [PostgreSQL DROP TABLE](https://www.postgresql.org/docs/current/sql-droptable.html)
- [MySQL DROP TABLE](https://dev.mysql.com/doc/refman/8.0/en/drop-table.html)
- [Database Backup Best Practices](https://www.sqlshack.com/sql-server-database-backup-best-practices/)

## Implementation Notes

- `DROP TABLE` / `DROP DATABASE` — Severity: Error
- `DROP INDEX` / `DROP VIEW` — Severity: Warning
- Handle `IF EXISTS` modifier
- Handle `CASCADE` / `RESTRICT` modifiers
- Consider context: migration files might legitimately use DROP
